readthedocs/vcs_support/backends/git.py

```diff
@@ -27,6 +21,29 @@ class Backend(BaseVCS):
     supports_branches = True
     fallback_branch = 'master'  # default branch
 
+    #: Regex pattern to match tags from output of `git show-ref --tags`
+    TAG_REGEX = re.compile(
+        r'''
+        ^\s*                        # any amount of whitespace
+        (?P<hash>[0-9a-f]{40,64})   # sha-1 or sha-256 hash
+        \s+                         # any amount of whitespace
+        (?P<tag>.*)                 # name of the tag, simplified to catch any char
+        (?:\n|$)                    # end of line or string
+        ''',
+        (re.VERBOSE | re.MULTILINE)
+    )
+    #: Regex pattern to match branch names from output of `git branch -r`. The
+    #: rules for branch names are a lot more specific than this, but not something
+    #: that should be maintained here.
+    BRANCH_REGEX = re.compile(
+        r'''
+        ^\s*                        # any amount of whitespace
+        (?P<branch>\w.+)            # an alpha-numeric character followed by anything
```

Comment:
So this will match all of `origin/HEAD -> origin/master` (from the docstring of the parse_branches), rather than splitting it. I assume that was why the CSV reader was used previously (I can only imagine why `.split(' ')` wasn't used).